src/main/kotlin/no/nav/syfo/client/dokarkiv/DokarkivClient.kt

```diff
@@ -1,18 +1,66 @@
 package no.nav.syfo.client.dokarkiv
 
+import io.ktor.client.call.*
+import io.ktor.client.features.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import net.logstash.logback.argument.StructuredArguments
 import no.nav.syfo.client.azuread.AzureAdClient
+import no.nav.syfo.client.dokarkiv.domain.JournalpostRequest
+import no.nav.syfo.client.dokarkiv.domain.JournalpostResponse
+import no.nav.syfo.client.httpClientDefault
+import no.nav.syfo.util.bearerHeader
 import org.slf4j.LoggerFactory
 
 class DokarkivClient(
     private val azureAdClient: AzureAdClient,
     private val dokarkivClientId: String,
+    dokarkivBaseUrl: String,
 ) {
-    suspend fun journalfor() {
-        log.info("Requesting access token from AzureAD")
-        azureAdClient.getAccessTokenForResource(dokarkivClientId)
+    private val journalpostUrl: String = "$dokarkivBaseUrl$JOURNALPOST_PATH"
+
+    private val httpClient = httpClientDefault()
+
+    suspend fun journalfor(
+        journalpostRequest: JournalpostRequest,
+    ): JournalpostResponse? {
+        val accessToken = azureAdClient.getAccessTokenForResource(dokarkivClientId)?.access_token
+        accessToken?.let { token ->
+            return try {
+                val response: HttpResponse = httpClient.post(journalpostUrl) {
+                    parameter(JOURNALPOST_PARAM_STRING, JOURNALPOST_PARAM_VALUE)
+                    header(HttpHeaders.Authorization, bearerHeader(token))
+                    accept(ContentType.Application.Json)
+                    contentType(ContentType.Application.Json)
+                    body = journalpostRequest
+                }
+                val journalpostResponse = response.receive<JournalpostResponse>()
+                COUNT_CALL_DOKARKIV_JOURNALPOST_SUCCESS.inc()
+                journalpostResponse
+            } catch (e: ClientRequestException) {
+                handleUnexpectedResponseException(e.response)
+            } catch (e: ServerResponseException) {
+                handleUnexpectedResponseException(e.response)
+            }
+        } ?: throw RuntimeException("Failed to Journalpost: No accessToken was found")
```

Comment:
"Failed to **journalfor** Journalpost ...  " ?

Reply:
Fikset.